builtin/mainmenu/tab_settings.lua

```diff
@@ -389,6 +423,32 @@ local function handle_settings_buttons(this, fields, tabname, tabdata)
 		core.setting_set("anisotropic_filter", "true")
 		ddhandled = true
 	end
+	
+	if fields["dd_resolution"] == "800 x 600 (4:3)" then
+		core.setting_set("screenW", "800")
+		core.setting_set("screenH", "600")
+		ddhandled = true
+	end
+	if fields["dd_resolution"] == "960 x 540 (16:9)" then
```

Comment:
You test the same field which isn't altered before the next condition, you need to replace if by else if.